### PR TITLE
Updates to accommodate pycparser update

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by simonlindholm** [comment]
Can you update stubs/ too similar to how m2c did it, and make sure mypy runs without errors?

File: stubs/pycparser/c_ast.py [hunk 1]
```diff
@@ -344,20 +360,22 @@ def __init__(self, coord: Optional[Coord] = None):
...
 
     def __init__(
         self,
         name: Optional[str],
         quals: List[str],
+        align: List[Alignas],
         storage: List[str],
         funcspec: List[str],
-        type: Union_[Type, "Struct", "Union", "Enum"],
-        init: Optional[Union_[Expression, "InitList"]],
```
Comment (simonlindholm) [inline stubs/pycparser/c_ast.py:360]: revert these changes (these fixes to Decl should really be mirrored to m2c, I must have forgot)

Comment (ethteck) [reply stubs/pycparser/c_ast.py:360]: just these two lines, or all changes to the Decl class?

Comment (simonlindholm) [reply stubs/pycparser/c_ast.py:360]: align is still needed, but revert the type and init changes, here and the `type` thing higher up
